### DISCUSSION TITLE
Unit and grammatical problems of elastic tensor

### DISCUSSION THREAD

**Discussion by biaogxb** [idaholab/moose#21282] [Q&A Modules: Solid mechanics] [answered]
Hello, I am coupling phase field and elastic energy, and I have encountered two problems:

**1**.I use the following formula to calculate cijkl in ComputeElasticityTensor，
![image](https://user-images.githubusercontent.com/95063954/173316435-594a6d3e-475e-46b6-abd5-061bdfec03fc.png)
and E（4.9Gpa） is the Young's modulus; v（0.36） is the Poisson's ratio，What should be the input unit for Young's modulus（Gpa or pa？） calculation in MOOSE？
I use Pa as the unit, and the code is as follows：
`[./Stiffness_matrix]
    type = ComputeElasticityTensor
    C_ijkl = '8.232e9 4.632e9 4.632e9 8.232e9 4.632e9 8.232e9 1.8e9 1.8e9 1.8e9'
    fill_method = symmetric9
  [../]`

**2**.I use ComputeVariableEigenstrain to calculate the strain, but I don't know how to fill in “eigen_base” ，The references I refer to for this part of the calculation are as follows：
the elastic stain tensor is 
![image](https://user-images.githubusercontent.com/95063954/173323282-028f0634-f225-4a4d-960c-cbb32dad7b24.png)
![image](https://user-images.githubusercontent.com/95063954/173322966-7d4d75fc-2094-4689-9c91-c40c69ea0319.png)
Whether this “eigen_base” is filled with the value of λ（1.2.3）, and whether the unit of the value of  λ（1.2.3）is pa or Gpa？

I use Gpa as the unit to fill in the following code. I don't know whether it is correct？
 `  [./eigen_strain]
    type = ComputeVariableEigenstrain
    eigen_base = '-0.866e-3 0 0 0 -0.773e-3 0 0 0 -0.529e-3'
    prefactor = h
    args = eta
    eigenstrain_name = 'eigenstrain'
  [../]`

Here is the original text of this article
[1-s2.0-S0013468618302809-main.pdf](https://github.com/idaholab/moose/files/8889857/1-s2.0-S0013468618302809-main.pdf)

Can someone help me? Thank you.

**Comment by biaogxb** [comment, marked as answer]
Because my length unit is micrometer, but Pa includes meter, so I need to multiply E-6 to convert the unit, right？
![image](https://user-images.githubusercontent.com/95063954/173506767-bf550d81-a359-44cf-9610-111e18ee532f.png)

**Reply by WilkAndy** [reply]
This is not necessarily true, and is probably false.  The fundamental point is to make all the units in your input file consistent (eg, do not use Pa somewhere and GPa somewhere else).  You may find the section "An essay on Pascals, kilograms and densities" in https://mooseframework.inl.gov/source/materials/PorousFlowSingleComponentFluid.html useful.  Although that essay is written in the context of PorousFlow, it may help clarify your thoughts, specifically around using Pascals rather than kg as your fundamental unit.

**Reply by biaogxb** [reply]
Thank you, I will study this！

**Reply by GiudGiud** [reply]
I'll add that the units in the input file have to be consistent with the mesh as well. The easiest way is to just use SI units for everything

**Reply by biaogxb** [reply]
thank you！I decided to use the SI for my input file.
